topo/topo_test.go

```diff
@@ -0,0 +1,135 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package topo
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	tfake "github.com/google/kne/api/clientset/v1beta1/fake"
+	tpb "github.com/google/kne/proto/topo"
+	"github.com/h-fam/errdiff"
+	kfake "k8s.io/client-go/kubernetes/fake"
+
+	"k8s.io/client-go/rest"
+)
+
+var (
+	opts []Option
```

Comment:
I don't think this is necessary, the fact that it is global and it gets a reset deferred after a single test function is overly complicated.

Its probably sufficient to instead change the block at the top of the test function to:

```
tf, err := tfake.NewSimpleClientset()
if err != nil {
    t.Fatalf("cannot create fake topology clientset")
}
opts := []Option{
    WithClusterConfig(&rest.Config{}),
    WithKubeClient(kfake.NewSimpleClientset()),
    WithTopoClient(tf),
}
```